code/modules/atmospherics/machinery/components/binary_devices/circulator.dm

```diff
@@ -1,24 +1,32 @@
 //node2, air2, network2 correspond to input
 //node1, air1, network1 correspond to output
 
-
 /obj/machinery/atmospherics/components/binary/circulator
 	name = "circulator/heat exchanger"
 	desc = "A gas circulator pump and heat exchanger."
-	icon_state = "circ1-off"
+	icon_state = "circ-off"
 
-	var/side = CIRC_LEFT
-	var/status = 0
+	var/active = FALSE
 
 	var/last_pressure_delta = 0
 	pipe_flags = PIPING_ONE_PER_TURF
 
 	anchored = TRUE
 	density = TRUE
+	var/const/hot = 0
+	var/const/cold = 1
+
+	var/mode = hot
+	var/obj/machinery/power/generator/generator

```

Comment:
How does this handle the generator being deleted?

Reply:
I added kill_circs() to the generators destroy(), so should handle it fine